### ISSUE TITLE
What is the point of setPosition

### ISSUE THREAD

**Issue by GuichiZhao** [ericdrowell/concrete#4]
I can not identify any meaning logic relation to `setPosition`
Is there anything yet to be implemented?

**Comment by ericdrowell** [comment]
ah yes - most people would probably not need it, but you can use it to improve performance for special situations.  For example if you have a complex rendering on a layer, but just need to move it around, you could use setPosition() to physically move the canvas element as opposed to constantly re rendering.  Alternatively, you could copy the canvas, and then re render the canvas pixels at different locations.

**Comment by ericdrowell** [comment]
you do have a point however, as most folks would probably never use it

**Comment by GuichiZhao** [comment]
I think the situation you mentioned is sort of related to what I want. 
I need to 'move around' the canvas. From the source code, `setPosition` looks like 'not implemented'. It do nothing but set `this.x`  `this.y`, and both of the two variables are not used afterward.
Could you provide more sample code to show how to use `setPosition`? 

Currently, I use `clip` to simulate 'move around' the whole canvas. It will be very nice if you can point out how to use `setPosition` 

Many thanks

**Comment by ericdrowell** [comment]
you are absolutely correct, setPosition() isn't actually moving the dom element.  Must have been something I was thinking about but missed.  setPosition() should also do:

this.viewport.container.style.marginLeft = x;
this.viewport.container.style.marginTop = y;

or something similar

**Comment by GuichiZhao** [comment]
Currently, I can not come up with a way to make `setPosition` meaningful. 
Because `setPosition` is a method of `Layer`, even if it can be inherited by `Scene` or `Hit` somehow. They both live in memory (except the scene instance belonging to the viewport). For most in-memory canvas, how to implement `setPosition`

Correct me If I understand things more wrong, my knowledge of canvas is limited

**Comment by marstave** [comment]
Just alter the render function and put render.x and render.y for the two 0,0 in drawImage :)

**Comment by GuichiZhao** [comment]
@marstave 
This is also what I thought where `position` can fit in :)

**Comment by nikrowell** [comment]
Hi Guys - 
Thinking about using **Concrete** for a lightweight game / easter egg on a site. I know this discussion is old, and not sure how much attention the library is being given, but I came across the same question on `x` and `y` layer positioning. It looks like this can be implemented by simply changing the `viewport.render` method:

```javascript
// scene.context.drawImage(layer.scene.canvas, 0, 0, layer.width, layer.height);
scene.context.drawImage(layer.scene.canvas, layer.x, layer.y, layer.width, layer.height);
```
